docs/PTLC.md

```diff
@@ -0,0 +1,539 @@
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`signature`:`signature`]
+   * [`u16`:`num_htlcs`]
+   * [`num_htlcs*signature`:`ptlc_signature`]
+   
+
+### Changes in the transaction and script format
+
+
+
+#### Offered and Received PTLC Outputs
+
+
+
+We use the same format for both offered and received PTLC outputs:
+
+
+
+```
+OP_DUP  OP_HASH160 <RIPEMD160(SHA256(revocationpubkey))> OP_EQUAL
+OP_IF 
+    OP_CHECKSIG
+OP_ELSE 
+    OP_2 <remotePtlcPubkey> <localPtlcPubkey> OP_2 OP_CHECKMULTISIG
+OP_ENDIF
+```
+
+
+### PTLC feature flag
+
+
+
+PTLC feature flag enables support of PTLC for node. Also it should be set in invoices to initiate PTLC payments. 
+
+Note, that `var_onion_optin` feature should be set as well to support PTLC.
+
+
+
+| Bits  | Name                             | Description                 | Context  | Dependencies      |
+|-------|----------------------------------|-----------------------------|----------|-------------------|
+| 58/59 | `ptlc`                           | Requires or supports PTLC   | IN9      | `var_onion_optin` |
+
+
+
+### PTLC Tagged Field (Invoices)
+
+
+
+The PTLC field gets populated if the node enables PTLC via setting PTLC feture flag.
+
+In this case `payment_hash` field is set to the SHA256 of the payment point.
+
+
+
+* (31): data_length 33. Payment point. Preimage of this provides proof of payment.
+
+
+
+### Onion TLV type
+
+Each hop should have its own point tweak, so it's encoded in the onion packet.
+
+1. type: 0x5555555555555 (next point tweak)
+2. data:
+    * [32*byte : next_point_tweak]
+
+
+
+#How to Run ad Test PTLC
+
+
+
+## Setting up a regtest network of Eclair nodes
+
+
+
+Download and install Bitcoin Core from https://bitcoin.org/en/download.
+
+
+
+Open Bitcoin Core config file in your favorite text editor (the file is located at `~/.bitcoin/bitcoin.conf` on Linux, or `~/Library/Application Support/Bitcoin` on Mac OS X) 
+
+and paste the configuration below.
+
+
+```
+regtest=1
+server=1
+txindex=1
+
+rpcuser=foo
+rpcpassword=bar
+
+zmqpubrawblock=tcp://127.0.0.1:29000
+zmqpubrawtx=tcp://127.0.0.1:29001
+```
+
+
+
+Start bitcoind
+
+
+
+```bash
+$ cd <Bitcoin core installation directory>/bin
+$ ./bitcoind
+```
+
+
+
+In another terminal window generate some test coins
+
+
+
+```bash
+$ cd <Bitcoin core installation directory>/bin
+$ ./bitcoin-cli getnewaddress
+# it will print a new address here, copy and paste it to the next commands
+$ ./bitcoin-cli generatetoaddress 100 <paste_the_address_here>
+$ ./bitcoin-cli generatetoaddress 10 <paste_the_address_here>
+```
+
+
+
+To build and run Eclair first you need to install Java: https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=hotspot
+
+
+
+Also, you'll need Maven to build Eclair, follow installation instructions here: https://maven.apache.org/install.html. 
+
+
+
+Open another terminal and clone Elair repository
+
+
+
+```bash
+$ cd
+$ git clone https://github.com/bitcoin-s/eclair.git
```

Comment:
do they need to checkout a specific branch? Do we want to merge this into our master branch?

Reply:
I'm going to merge it to the default branch, which is `develop`.